### PR TITLE
Add a CONTRIBUTING file to the project

### LINKED ISSUE #240: Contributing guidelines
There were a couple assumptions made with #239:
- create pull request on development branch
- run ``cargo test --all``

Adding to CHANGES.md slipped my mind. A CONTRIBUTING.md would make preferences of what to add and run before opening a pull request clear.

By the way, love the direction with Quicksilver :)

### PR DESCRIPTION
**Pull request by ryanisaacg**
Resolves #240

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+The third category of issue is a feature request.
+Quicksilver is an evolving project that can definitely use a lot more features, and the project is very open to suggestions.
+However, there is a certain category of feature that (generally) won't be considered for work: anything that will only work on one of desktop or web.
+A firm goal of quicksilver is unsurprising behavior when taking a web-first project and compiling for desktop and vice-versa.
+Also, there's no guarantee anyone else will be up to working on your feature request; if that's the case, then feel free to [open a PR](#prs).
+
+
+## PRs
+
+The first rule of PRs to Quicksilver to generally try to make only a few changes in a single PR.
```
Comment (memoryruins) [inline CONTRIBUTING.md]: This line might be missing an ``is`` in "... Quicksilver ``is`` to generally ...".
If preferred, the entire line could be shortened to:
"Generally, try to make only a few changes in a single PR to Quicksilver."

Comment (ryanisaacg) [reply CONTRIBUTING.md]: Thanks, fixed

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+### Documenting your changes
+
+- Add an entry in `CHANGES.md`, which can generally be the summary(s) of your git commits
+- If you changed the public API, note [BREAKING] in front of your change
+- Additionally, make sure you didn't break the code snippet in the website, the README, or `src/lib.rs`. If you did, make sure to update them.
+
+### Testing your changes
+
+- Make sure to run `cargo test` which will both verify the tests pass and also make sure the crate and all examples compile
+- Make sure to run `cargo +nightly check --target wasm32-unknown --examples` which will verify that the crate and all examples compile for web
```
Comment (memoryruins) [inline CONTRIBUTING.md]: The command produced the following error:
```
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names --target wasm32-unknown --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro` (exit code: 101)
```
Adding an extra ``-unknown`` made rustc happy:
``cargo +nightly check --target wasm32-unknown-unknown --examples``

Comment (ryanisaacg) [reply CONTRIBUTING.md]: Yup sorry about that typo! Pushed a fix.

**Pull request merged**